firebase-common/src/androidMain/kotlin/dev/gitlive/firebase/_decoders.kt

```diff
@@ -10,19 +10,9 @@ import kotlinx.serialization.descriptors.SerialDescriptor
 import kotlinx.serialization.descriptors.StructureKind
 
 actual fun FirebaseDecoder.structureDecoder(descriptor: SerialDescriptor): CompositeDecoder = when(descriptor.kind) {
-    StructureKind.CLASS, StructureKind.OBJECT -> when {
-        value is Map<*, *> ->
-            FirebaseClassDecoder(value.size, { value.containsKey(it) }) { desc, index ->
-                value[desc.getElementName(index)]
-            }
-        value != null && value::class.qualifiedName == "com.google.firebase.Timestamp" -> {
```

Comment:
there is no need to use reflection if `FirebaseNullableTimestampSerializer`, `TimestampNullableSerializer` or other serializers are used (see below)